pkgs/development/compilers/terra/default.nix

```diff
@@ -1,61 +1,70 @@
 { lib, stdenv, fetchurl, fetchFromGitHub
-, llvmPackages, ncurses, lua
+, llvmPackages, ncurses, cmake, libxml2, symlinkJoin
 }:
 
 let
-  luajitArchive = "LuaJIT-2.0.5.tar.gz";
+  luajitRev = "9143e86498436892cb4316550be4d45b68a61224";
+  luajitArchive = "LuaJIT-${luajitRev}.tar.gz";
   luajitSrc = fetchurl {
-    url    = "http://luajit.org/download/${luajitArchive}";
-    sha256 = "0yg9q4q6v028bgh85317ykc9whgxgysp76qzaqgq55y6jy11yjw7";
+    url = "https://github.com/LuaJIT/LuaJIT/archive/${luajitRev}.tar.gz";
+    sha256 = "0kasmyk40ic4b9dwd4wixm0qk10l88ardrfimwmq36yc5dhnizmy";
   };
 in
 stdenv.mkDerivation rec {
   pname = "terra";
-  version = "1.0.0pre1175_${builtins.substring 0 7 src.rev}";
+  version = "1.0.0-beta3";
 
   src = fetchFromGitHub {
-    owner  = "zdevito";
+    owner  = "terralang";
     repo   = "terra";
-    rev    = "ef6a75ffee15a30f3c74f4e6943851cfbc0fec3d";
-    sha256 = "0aky17vbv3d9zng34hp17p9zb00dbzwhvzsdjzrrqvk9lmyvix0s";
+    rev    = "release-${version}";
+    sha256 = "15ik32xnwyf3g57jvvaz24f6a18lv3a86341rzjbs30kd5045qzd";
   };
 
-  nativeBuildInputs = [ lua ];
-  buildInputs = with llvmPackages; [ llvm clang-unwrapped ncurses ];
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [
+    (symlinkJoin {
+      name = "llvmClangMerged";
+      paths = with llvmPackages; [ llvm clang-unwrapped ];
```

Comment:
This is going to need to be tweaked in response to https://github.com/NixOS/nixpkgs/commit/7869d1654517c028aa76fc1dedc9b5ac301a867a, and the splitting of every LLVM package into multiple outputs. It looks like including `.out`, `.lib`, and `.dev` for each should be sufficient.